AlgorithmFactory/Python/Wrappers/AlgorithmPythonWrapper.cs

```diff
@@ -100,7 +105,29 @@ public AlgorithmPythonWrapper(string moduleName)
 
                             _onOrderEvent = pyAlgorithm.GetAttr("OnOrderEvent");
 
-                            IsOnEndOfDayImplemented = pyAlgorithm.GetPythonMethod("OnEndOfDay") != null;
+                            PyObject endOfDayMethod = pyAlgorithm.GetPythonMethod("OnEndOfDay");
+                            if (endOfDayMethod != null)
+                            {
+                                // Since we have a EOD method implemented
+                                // Determine which one it is by inspecting its arg count
+                                int argCount;
+                                var pyArgCount = PythonEngine.ModuleFromString(Guid.NewGuid().ToString(),
+                                "from inspect import signature\n" +
+                                    "def GetArgCount(method):\n" +
+                                    "   return len(signature(method).parameters)\n"
+                                ).GetAttr("GetArgCount").Invoke(endOfDayMethod);
```

Comment:
Nice! This could be an extension method too?
What would happen if both are implemented?

Reply:
If both are implemented, the latest defined takes precedence. It is Python that does this, basically overwriting the first definition with this one. Meaning no two functions can share a name with different sigs